test/cases/edge.sh

```diff
@@ -0,0 +1,360 @@
+  do
+    RESULT=$($CURLCMD -H "$HEADER" --request GET "$BASEURL/composes/$COMPOSE_ID")
+    EXIT_CODE=$(get_exit_code "$RESULT")
+    [[ $EXIT_CODE == 200 ]]
+
+    COMPOSE_STATUS=$(get_response "$RESULT" | jq -r '.image_status.status')
+    UPLOAD_STATUS=$(get_response "$RESULT" | jq -r '.image_status.upload_status.status')
+
+    case "$COMPOSE_STATUS" in
+      # "running is kept here temporarily for backward compatibility"
+      "running")
+        ;;
+      # valid status values for compose which is not yet finished
+      "pending"|"building"|"uploading"|"registering")
+        ;;
+      "success")
+        [[ "$UPLOAD_STATUS" = "success" ]]
+        break
+        ;;
+      "failure")
+        echo "Image compose failed"
+        exit 1
+        ;;
+      *)
+        echo "API returned unexpected image_status.status value: '$COMPOSE_STATUS'"
+        exit 1
+        ;;
+    esac
+    sleep 30
+  done
+}
+
+# Download image from aws s3
+function download_image() {
+    RESULT=$($CURLCMD -H "$HEADER" --request GET "$BASEURL/composes/$COMPOSE_ID")
+    RESULT_URL=$(get_response "$RESULT" | jq -r '.image_status.upload_status.options.url')
+    echo "Downloading image, URL is: $RESULT_URL, save it as $1"
+    $CURLCMD "$RESULT_URL" --output "$1"
+}
+
+# Test result checking
+check_result () {
+    greenprint "🎏 Checking for test result"
+    if [[ $RESULTS == 1 ]]; then
+        greenprint "💚 Success"
+    else
+        greenprint "❌ Failed"
+        after_test
+        exit 1
+    fi
+}
+
+# Wait for the ssh server up to be.
+function wait_for_ssh() {
+    SSH_STATUS=$(sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" admin@"${1}" '/bin/bash -c "echo -n READY"')
+    if [[ $SSH_STATUS == READY ]]; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+# Run before test begin to prepare test environment
+function before_test() {
+    # Check service status
+    READY=0
+    for RETRY in {1..10};do
+    curl --fail -H "$HEADER" "http://$ADDRESS:$PORT/ready" && {
+        READY=1
+        break
+    }
+    echo "Port $PORT is not open. Waiting...($RETRY/10)"
+    sleep 1
+    done
+
+    [ "$READY" -eq 1 ] || {
+    echo "Port $PORT is not open after retrying 10 times. Exit."
+    exit 1
+    }
+
+    # ansible is not in RHEL repositories, enable EPEL and install ansible manually.
+    sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+    sudo dnf install -y ansible
+
+    # Start libvirtd and test it.
+    greenprint "🚀 Starting libvirt daemon"
+    sudo systemctl start libvirtd
+    sudo virsh list --all > /dev/null
+
+    if ! sudo virsh net-info integration > /dev/null 2>&1; then
+        sudo virsh net-define ${IMAGE_BUILDER_TEST_DATA}/edge/integration.xml
+        sudo virsh net-start integration
+    fi
+
+    # Allow anyone in the wheel group to talk to libvirt.
+    greenprint "🚪 Allowing users in wheel group to talk to libvirt"
+    WHEEL_GROUP=wheel
+    if [[ $ID == rhel ]]; then
```

Comment:
Since you did not run `source /etc/os-release` in this script, `$ID` does not exist.

Reply:
Thanks for pointing out this, added a line to source /etc/os-release.